### PR TITLE
[wip] Load proposal data from sqlite, instead via chef api

### PR DESCRIPTION
**Pull request by bkutil**
Needs testing, just checking if this is the correct approach.

cc: @vuntz @aspiers @tboerger

### CODE REVIEW

File: crowbar_framework/app/models/service_object.rb [hunk 1]
```diff
@@ -769,7 +769,8 @@ def proposal_commit(inst, in_queue = false, validate_after_save = true)
         # Put mark on the wall
         prop["deployment"][@bc_name]["crowbar-committing"] = true
         save_proposal!(prop, :validate_after_save => validate_after_save)
-        response = active_update prop.raw_data, inst, in_queue
+        sqlite_prop = Proposal.where(:barclamp => @bc_name, :name => inst).first || raise("Failed to find proposal #{@bc_name}-#{inst}.")
```
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: Localization of error message ;-)

**Comment by vuntz** [comment]
You might want to clarify the commit message (to mention that this is about committing).

Would it be possible to remove the call at https://github.com/crowbar/barclamp-crowbar/pull/1260/files#diff-0be8ea1f05c9441a59bb2b7ac65b1c62R760 ? Because in the end, I think this is really what we should target here.

**Comment by bkutil** [comment]
@vuntz we change the proposal afterwards (save_proposal!) and the changes would be made to `Proposal` instead of `ProposalObject` record. But if #1261 is merged, the models are synced both ways, so it should work then.

**Comment by aspiers** [comment]
Agree with @vuntz, explanations belong in the commit message not as comments in the PR.

**Comment by bkutil** [comment]
Updated the commit message. Merging still depends on #1261.

**Comment by tboerger** [comment]
Failing tests :(

**Comment by tboerger** [comment]
Awesome when the tests work! :)

**Comment by bkutil** [comment]
@tboerger they should pass now :)

**Comment by tboerger** [comment]
:+1:

**Comment by vuntz** [comment]
+1

**Pull request merged**